go-controller/pkg/cni/helper_windows.go

```diff
@@ -123,8 +123,16 @@ func (pr *PodRequest) ConfigureInterface(namespace string, podName string, macAd
 	endpointName := fmt.Sprintf("%s_%s", namespace, podName)
 
 	defer func() {
-		// Delete the endpoint in case CNI fails
+		// Delete the OVS port and endpoint in case CNI fails
```

Comment:
Would it be better to create a separate function for cleanup and then call it here and in PlatformSpecificCleanup since it's almost the same?

Reply:
But would that function be called if you do ``ovs-vsctl del-port`` but you don't have any port added with ``ovs-vsctl add-port``? Since CNI creates the HNSEndpoints, it should also be responsible for managing the deletion or at least checking that it doesn't leak.